src/components/menus/base-equipment-menu.js

```diff
@@ -83,15 +109,29 @@ const BaseEquipmentMenu = ({
             {/* menus for equipment other than substation and voltage level */}
             {equipmentType !== equipments.substations &&
                 equipmentType !== equipments.voltageLevels && (
-                    <ItemViewInSpreadsheet
-                        key="ViewOnSpreadsheet"
-                        equipmentType={equipmentType}
-                        equipmentId={equipmentId}
-                        itemText={intl.formatMessage({
-                            id: 'ViewOnSpreadsheet',
-                        })}
-                        handleViewInSpreadsheet={handleViewInSpreadsheet}
-                    />
+                    <>
+                        <ItemViewInSpreadsheet
+                            key="ViewOnSpreadsheet"
+                            equipmentType={equipmentType}
+                            equipmentId={equipmentId}
+                            itemText={intl.formatMessage({
+                                id: 'ViewOnSpreadsheet',
+                            })}
+                            handleViewInSpreadsheet={handleViewInSpreadsheet}
+                        />
+                        {equipmentType === equipments.generators && (
```

Comment:
Why is it nested under `{equipmentType !== equipments.substations && equipmentType !== equipments.voltageLevels` ?
Couldn't it be on its own ?

Reply:
yes, i move it for it's own bloc